### DISCUSSION TITLE
Let's use keys instead of parameters

### DISCUSSION THREAD

**Discussion by Alerinos** [SixLabors/ImageSharp.Web#150] [Ideas] [answered]
I got the idea that instead of declaring parameters with each picture of the type
```
?format=webp&width=515&height=290
```
You could use keys of type
```
?key=abcd123
```
Each key would be declared in the settings and its value would be assigned to specific settings. Thanks to this, we have the security that someone will not be able to attack us and change any image formats.
What do you think about such a solution? In addition, the link looks nice and nobody will come up with the idea to make our lives miserable.
It also makes sense in faster work, instead of remembering all the settings we can set
```
?key=small-image
?key=large-image
?key=avatar
```

**Comment by JimBobSquarePants** [comment, marked as answer]
You can already do this by implementing and registering your own [`IRequestParser`](https://github.com/SixLabors/ImageSharp.Web/blob/d0559520245dffc484cd7fc77ae35fd20f0ff3b9/src/ImageSharp.Web/Commands/IRequestParser.cs#L12)

You would map the incoming requests to a series of commands that then get processed.
